### PR TITLE
Fix issue with worker function not being executed

### PR DESCRIPTION
**Pull request by mantono**
- Remove `src/main/resources/logback.xml`, since it should not be included in a library
- Add additional logging, to make it clearer when something "breaks" and a worker is not run because of thread starvation
- Do **not** default to GlobalScope as default CoroutineScope, instead create an internal CoroutineScope that can be overriden with a custom CoroutineScope
- Set CoroutineScope for worker in builder function rather than _start_ function.
- Update Kotlin coroutines to version 1.2.0
- Make formatting changes to be more consistent with indenting and specify return time for functions when it is not obvious from context

### CODE REVIEW

File: src/main/kotlin/engine/kafka_one/KafkaConsumerActor.kt [hunk 1]
```diff
@@ -42,19 +42,27 @@ private fun <T, U> commitFinishedJobs(c: KafkaConsumer<T, U>,
...
             logger.info { "Adding ${it.count()} new tasks from Kafka" }
+        } else {
+            logger.debug { "Adding no new tasks from Kafka" }
         }
+        logger.debug { "Size outQueue: ${outQueue.size}, remaining capacity: ${outQueue.remainingCapacity()}" }
         outQueue.offerAll(it) to jobStatuses.addJobs(it)
     }
 
-private fun <T> BlockingQueue<T>.offerAll(xs: Iterable<T>) = xs.dropWhile { offer(it) }
+private fun <T> BlockingQueue<T>.offerAll(xs: Iterable<T>): List<T> = xs.dropWhile { offer(it) }
```
Comment (Torbacka) [inline src/main/kotlin/engine/kafka_one/KafkaConsumerActor.kt:60]: Nice, love when people set things like this explicit.

**Review by Torbacka — APPROVED** [review]
Nice with some more debug logging

File: src/main/kotlin/WorkerBuilder.kt [hunk 2]
```diff
@@ -27,7 +32,8 @@ data class WorkerBuilder<T> private constructor(
     private val opts: Map<String, Any> = emptyMap(),
     private val topics: List<String> = emptyList(),
     private val engine: ConsumerActorFactory = KafkaConsumerActorFactory,
-    private val interceptors: List<WorkerInterceptor> = emptyList()
+    private val interceptors: List<WorkerInterceptor> = emptyList(),
+    private val scope: CoroutineScope = createDefaultScope()
```
Comment (bonahona) [inline src/main/kotlin/WorkerBuilder.kt]: The variable name is vague. What scope does it refer to. Change it to reflect its a coroutine scope.

File: src/main/kotlin/WorkerBuilder.kt [hunk 3]
```diff
@@ -38,13 +44,14 @@ data class WorkerBuilder<T> private constructor(
     fun groupId(id: String) = option("group.id", id)
     fun option(k: String, v: Any) = options(mapOf(k to v))
     fun options(newOpts: Map<String, Any>) = copy(opts = opts + newOpts)
+    fun scope(scope: CoroutineScope): WorkerBuilder<T> = copy(scope = scope)
```
Comment (bonahona) [inline src/main/kotlin/WorkerBuilder.kt]: Again, the name is vague and ambigious. Naming it coroutine scope or maybe even something that more reflects its a executing couroutine scope for the worker would be better.

File: src/main/kotlin/WorkerBuilder.kt [hunk 4]
```diff
@@ -38,13 +44,14 @@ data class WorkerBuilder<T> private constructor(
...
     fun setEngine(e: ConsumerActorFactory): WorkerBuilder<T> = copy(engine = e)
 
     fun install(i: WorkerInterceptor): WorkerBuilder<T> = copy(interceptors = interceptors.toMutableList().also{ it.add(i)})
 
     fun getInterceptors() = interceptors
 
-    fun start(scope: CoroutineScope = GlobalScope){
```
Comment (bonahona) [inline src/main/kotlin/WorkerBuilder.kt:47]: This should never have been here in the first place. Nice fix.

File: src/main/kotlin/WorkerBuilder.kt [hunk 5]
```diff
@@ -72,3 +79,17 @@ data class WorkerBuilder<T> private constructor(
     }
 }
 
+private const val THREAD_POOL_CORE_SIZE: Int = 1
+private const val THREAD_POOL_MAX_SIZE: Int = 2
+private const val THREAD_POOL_KEEP_ALIVE_TIME_SECONDS: Long = 30
+
+private fun createDefaultScope(): CoroutineScope {
+    val dispatcher: CoroutineDispatcher = ThreadPoolExecutor(
+        THREAD_POOL_CORE_SIZE,
```
Comment (bonahona) [inline src/main/kotlin/WorkerBuilder.kt]: Im confused by this name. I know it is what kotlin calls it.
Comments to explain the different constants might be a useful addition.

File: src/main/kotlin/engine/kafka_one/KafkaConsumerActor.kt (hunk 1 above)
Comment (bonahona) [reply src/main/kotlin/engine/kafka_one/KafkaConsumerActor.kt:60]: Agreed. Even if this is a standard Kotlin operation on a collection, explicit typing are nice.

**Review by bonahona — CHANGES_REQUESTED** [review]

File: src/main/kotlin/WorkerBuilder.kt (hunk 3 above)
Comment (mantono) [reply src/main/kotlin/WorkerBuilder.kt]: Sure!

File: src/main/kotlin/WorkerBuilder.kt (hunk 4 above)
Comment (mantono) [reply src/main/kotlin/WorkerBuilder.kt:47]: Yeah, that one is one me.

File: src/main/kotlin/WorkerBuilder.kt (hunk 5 above)
Comment (mantono) [reply src/main/kotlin/WorkerBuilder.kt]: Or maybe I should just rename the constant so a comment is not needed? :thinking:

Comment (bonahona) [reply src/main/kotlin/WorkerBuilder.kt]: Either works, as long as the purpose of the variable / constant is clear from just looking at it.

**Review by bonahona — APPROVED** [review]

**Pull request merged**